### DISCUSSION TITLE
Move all solidjs modules under `@solidjs` npm scope

### DISCUSSION THREAD

**Discussion by trivikr** [solidjs/solid#611] [Ideas]
This idea was originally discussed on [Discord Thread](https://discord.com/channels/722131463138705510/877563445107654667).

The core module `solid-js` can be republished under say `@solidjs/solid`, and there would be a standard followed across GitHub and Npm.

You can also republish:
* `solid-js/web` under `@solidjs/web`
* `solid-ssr` under `@solidjs/ssr`
* `solid-element` under `@solidjs/element`
* `solid-app-router` under `@solidjs/app-router` or `@solidjs/router`
* `solid-styled-components` to `@solidjs/styled-components`

Community maintained packages (like `@amoutonbrady/solid-heroicons`) can be also be published under `@solidjs/heroicons` if maintainers are ok with it.

**Comment by trivikr** [comment]
Update as of 08/19:
* Core maintainers @ryansolid and @davedbase are onboard to the idea.
* The org name `solidjs` is not available on npm.
* Npm currently has [a freeze on organization transfers for squatted/abandoned packages](https://docs.npmjs.com/policies/disputes#when-not-to-use-this-process).
* Created a [discussion thread on official npm feedback channel](https://github.com/npm/feedback/discussions/534) for timeline on when process is going to be updated.
* Sent an email to npm Support requesting for timeline. They responded stating that the request for `solidjs` org name is under review, and they'll get back.

**Reply by ryansolid** [reply]
I want to put out there I'm generally onboard with this one. I've noticed a lot of frameworks still have the core package not under the organization.. but give what we have is `solid-js` and not even `solidjs`or `solid` I'm probably onboard. Everything else completely makes sense.

Thanks for taking initiative.

**Reply by trivikr** [reply]
>  I've noticed a lot of frameworks still have the core package not under the repo

Did you mean a GitHub repository?

This initiative is just for publishing SolidJS projects under same npm scope `@solidjs`
The individual repositories can be managed separately, or in a monorepo, or both.

At AWS, we manage [AWS SDK for JavaScript (v3)](https://github.com/aws/aws-sdk-js-v3) in a monorepo on GitHub as the individual modules are standardized.
The projects under npm [@hapi](https://www.npmjs.com/search?q=%40hapi) scope are managed in individual repositories.

**Reply by ryansolid** [reply]
Edited ... I meant under the npm org. Like you `npm i vue` or `svelte` not `@vue/vue`. I think it's fine anyway.

**Reply by rturnq** [reply]
I believe Ryan is talking about keeping the the core package out of the npm org:
```js
import { ... } from 'solid-js'
```
vs
```js
import { ... } from '@solidjs/solid'
```
I think this makes sense. It is less typing and to my eyes feels a little more official. I think people would generally expect a bare name like this. I'm not sure how `solid-js/web` and `solid-js/store` should work as they are not separate packages today.

**Reply by trivikr** [reply]
Got it. I created a separate [Discord thread](https://discord.com/channels/722131463138705510/722131463889223772) on getting ownership of abandoned package [solid](https://www.npmjs.com/package/solid).
We should follow up after getting `solidjs` org on npm, or trademark - whichever comes first.

**Reply by davedbase** [reply]
Well to be honest my personal opinion is the org would be nicer and better suited as we grow as a community. Optics here personally matter to me, but for others it don't matter at all.

The argument about shorter and less typing makes sense, but isn't that a minor issue with the quality of IDEs we have these days that autofill?

I could argue that with `@solidjs` or `@solid` as an org, it's is easier to recognise "official" Solid packages. Anything published as the org is vetted and approved by the core. That doesn't mean much but in my eyes it just feels cleaner. It's also only a marginally more reasonable argument than typing less. In the future there might be reasons to "tag" an entire org and code published on it? Thinking about security possibly, being able to mark a namespace vs individual packages is convenient?

Generally TBH these are all pretty moot points. Depends on npm getting their act together and then it would be Ryan's decision regardless.

**Reply by ryansolid** [reply]
To be clear it seems more likely to take the org than it does to oust an existing package. So I'm fine moving under the org if we get everything tidy.

**Reply by trivikr** [reply]
Suggestion from Discord: Use `@solidjs/core` instead of `@solidjs/solid` for `solid-js`.

Similar to other examples:
* [@babel/core](https://www.npmjs.com/package/@babel/core) (29M weekly downloads)
* [@jest/core](https://www.npmjs.com/package/@jest/core) (11M weekly downloads)
* [@popperjs/core](https://www.npmjs.com/package/@popperjs/core) (4.2M weekly downloads)
* [@storybook/core](https://www.npmjs.com/package/@storybook/core) (2.8M weekly downloads)
* [@angular/core](https://www.npmjs.com/package/@angular/core) (2.5M weekly downloads)
* [@material-ui/core](https://www.npmjs.com/package/@material-ui/core) (2M weekly downloads)

**Reply by ryansolid** [reply]
Angular is the only framework in that list.. let's see @stencil/core. Oh know what does suck the module subnesting we do..

`@solidjs/core/web` `@solidjs/core/stores` Hmm.. timing of doing this is pretty critical. Honestly I don't think we could pull this sort of move until a major version change. Anything else is fine since nothing else is 1.0.

**Reply by trivikr** [reply]
> `@solidjs/core/web` `@solidjs/core/stores`

I would suggest using `@solidjs/web` and `@solidjs/stores` instead.

> Honestly I don't think we could pull this sort of move until a major version change.

True. Using scope should be reserved for v2.0.0. May be plan it for early 2022?

**Reply by ryansolid** [reply]
Yeah currently they are part of the same repo. I always liked how Preact kept that tight. I've always modelled things closer to them than say Angular.

Yeah we'll see. I don't want to spend a major version until there is a good reason for breaking changes. I imagine there will be 2 more minor versions this fall. We will see where things are after that. Honestly if everything moved except the core package for now we are still in really good shape.

**Reply by orenelbaum** [reply]
> > `@solidjs/core/web` `@solidjs/core/stores`
> 
> I would suggest using `@solidjs/web` and `@solidjs/stores` instead.

I like this. I think that from a user perspective having `web` its own thing, separate from the reactivity modules, would make more sense.

**Comment by trivikr** [comment]
Update as of 01/23: The `solidjs` org name on npm was transferred to me in response to support request to npm.
I've transferred it to @ryansolid - creator of SolidJS.

**Reply by trivikr** [reply]
As per previous discussions, moving to `solidjs` scope will mostly be prioritized for next major version bump, i.e. 2.0
The SolidJS core team may consider publishing packages under both `solid-*` and `@solidjs/*` names prior to 2.0 release.
